pulp_ansible/app/serializers.py

```diff
@@ -764,6 +967,57 @@ class Meta:
         )
 
 
+class CollectionVersionSigstoreSignatureSerializer(NoArtifactContentUploadSerializer):
+    """
+    A serializer for Sigstore signature models.
+    """
+    sigstore_bundle = serializers.JSONField(
+        help_text=_("A Sigstore bundle used for offline verification."),
+        allow_null=True,
+    )
+    signed_collection = DetailRelatedField(
+        help_text=_("The content this signature is pointing to."),
+        view_name_pattern=r"content(-.*/.*)-detail",
+        queryset=CollectionVersion.objects.all(),
+    )
+    sigstore_signing_service = RelatedField(
+        help_text=_("The signing service used to sign the collections"),
+        view_name="sigstore-signing-services-detail",
+        read_only=True,
+        allow_null=True,
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Ensure `file` field is required."""
+        super().__init__(*args, **kwargs)
+        self.fields["file"].required = True
+
+    def validate(self, data):
+        """
+        Verify the signature is valid before creating it.
+        """
+        if not "repository" in data:
+            raise serializers.ValidationError(
+                "'repository' field not set"
+            )
```

Comment:
I think you can remove this and just make the `repository` field required in the `init` like the `file` field.